pages/k8s/openstack-integration.md

```diff
@@ -51,6 +51,111 @@ relations:
+API server itself.
+
+In either case, the load balancers can optionally have floating IPs (FIPs) attached to them to
+allow for external access.
+
+<div class="p-notification--caution">
+  <p markdown="1" class="p-notification__response">
+    <span class="p-notification__status">Note:</span>
+For security reasons, the SGs automatically managed by Juju will not by default allow traffic into
+the nodes from external networks which can otherwise reach the FIPs. The easiest way to allow this
+is to add a rule to the model SG (named `juju-<model UUID>`) to allow ingress traffic from the FIP
+network, according to your security and network traffic policy and needs. Alternatively, you could
+create a separate SG to manage the rule(s) across multiple models or controllers.<br/>
+<br/>
+Manual SG intervention will also be required if you wish to have the Amphora instances in a
+different subnet from the node instances, since you will need to allow at least traffic on the
+NodePort range (30000-32767) from the Amphorae into the nodes.
+  </p>
+</div>
+
+#### LoadBalancer-type Pod Services
+
+To use Octavia for `LoadBalancer` type services in the cluster, you will also need to set the
+`subnet-id` config to the appropriate tenant subnet where your nodes reside, and if desired, the
+`floating-network-id` config to whatever network you want FIPs created in.  See the [Charm config
+docs][charm-config] for more details.
+
+As an example of this usage, this will create a simple application, scale it to five pods,
+and expose it with a `LoadBalancer`-type `Service`:
+
+```bash
+kubectl create deployment hello-world --image=gcr.io/google-samples/node-hello:1.0
+kubectl scale deployment hello-world --replicas=5
+kubectl expose deployment hello-world --type=LoadBalancer --name=hello --port 8080
+```
+
+You can verify that the application and replicas have been created with:
+
+```bash
+kubectl get deployments hello-world
+```
+
+Which should return output similar to:
+
+```bash
+NAME              READY   UP-TO-DATE   AVAILABLE   AGE
+hello-world      5/5               5                            5             2m38s
+```
+
+To check that the service is running correctly:
+
+```bash
+kubectl get service hello
+```
+
+...which should return output similar to:
+
+```yaml
+NAME    TYPE           CLUSTER-IP       EXTERNAL-IP   PORT(S)          AGE
+hello   LoadBalancer   10.152.183.136   202.49.242.3  8080:32662/TCP   2m
+```
+
+You can see that the External IP is now in front of the five endpoints of the
+example deployment. You can test the ingress address:
+
+```bash
+curl  http://202.49.242.3:8080
+```
+```
+Hello Kubernetes!
+```
+
+
+#### API Server Load Balancer
+
 If desired, the openstack-integrator can also replace kubeapi-load-balancer and create a native OpenStack
 load balancer for the Kubernetes API server, which both reduces the number of machines required and is
```

Comment:
I know it's not part of the review, but the 'both' in the above service looks superfluous?